src/main/java/it/eng/idsa/businesslogic/service/impl/DapsTokenProviderServiceImpl.java

```diff
@@ -64,7 +67,7 @@ public String provideToken() {
 	
 	@EventListener(ApplicationReadyEvent.class)
 	public void fetchTokenOnStartup() {
-		if (fetchTokenOnStartup && StringUtils.isBlank(cachedToken)) {
+		if ((fetchTokenOnStartup && isEnabledDapsInteraction) && StringUtils.isBlank(cachedToken)) {
```

Comment:
Nice one, not to fetch token if daps disabled - might not work if keystore is not provided